ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -266,3 +266,86 @@ module Host_requires_reboot = struct
         Unixext.touch_file Xapi_globs.requires_reboot_file
       )
 end
+
+module Configuration = struct
+
+  let make_initiatorname_config iqn hostname =
+    (* CA-18000: there is a 30 character limit to the initiator when talking to
+       Dell MD3000i filers, so we limit the size of the initiator name in all cases *)
+    let hostname_chopped =
+      if String.length hostname > 30
+      then String.sub hostname 0 30
+      else hostname
+    in
+    Printf.sprintf
+      "InitiatorName=%s\nInitiatorAlias=%s\n"
+      iqn hostname_chopped
+
+  let set_initiator_name iqn =
+    let hostname = Unix.gethostname () in
+    let config_file = make_initiatorname_config iqn hostname in
+    Unixext.write_string_to_file !Xapi_globs.iscsi_initiator_config_file config_file
+
+  let set_multipathing enabled =
+    let flag = !Xapi_globs.multipathing_config_file in
+    if enabled then Unixext.touch_file flag
+    else begin
+      if Sys.file_exists flag then Sys.remove flag
```

Comment:
I'm way of `Sys`... we cannot catch the errors properly when using it. Can we do this using Unix and maybe handle potential exceptions with internal_error?

Reply:
Replaced with Unixext.unlink_safe.

Reply:
Thanks!